cmd/revad/svcs/httpsvcs/ocdavsvc/ocdavsvc.go

```diff
@@ -268,32 +275,11 @@ func (s *svc) setHandler() {
 				}
 			}
 		}
+		log.Warn().Msg("resource not found")
 		w.WriteHeader(http.StatusNotFound)
 	})
 }
 
-func (s *svc) getConn() (*grpc.ClientConn, error) {
-	if s.conn != nil {
-		return s.conn, nil
-	}
-
-	conn, err := grpc.Dial(s.storageProviderSvc, grpc.WithInsecure())
-	if err != nil {
-		return nil, err
-	}
-
-	return conn, nil
-}
-
 func (s *svc) getClient() (storageproviderv0alphapb.StorageProviderServiceClient, error) {
-	if s.client != nil {
-		return s.client, nil
-	}
-
-	conn, err := s.getConn()
-	if err != nil {
-		return nil, err
-	}
-	s.client = storageproviderv0alphapb.NewStorageProviderServiceClient(conn)
-	return s.client, nil
+	return pool.GetStorageProviderServiceClient(s.gatewaySvc)
```

Comment:
this always gives me 
```
4:24PM ERR svcs/grpcsvcs/interceptors/recovery/recovery.go:50 > runtime error: invalid memory address or nil pointer dereference pid=3896864 pkg=grpcserver stack="goroutine 119 [running]:\nruntime/debug.Stack(0x0, 0x0, 0x0)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0xa1\ngithub.com/cs3org/reva/cmd/revad/svcs/grpcsvcs/interceptors/recovery.recoveryFunc(0x147f960, 0xc000596ea0, 0x10a3e40, 0x1c07850, 0x0, 0x0)\n\t/home/jfd/Repositories/nexus/build/reva/src/cmd/revad/svcs/grpcsvcs/interceptors/recovery/recovery.go:48 +0x42\ngithub.com/grpc-ecosystem/go-grpc-middleware/recovery.recoverFrom(0x147f960, 0xc000596ea0, 0x10a3e40, 0x1c07850, 0x1220968, 0x0, 0x0)\n\t/home/jfd/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/recovery/interceptors.go:51 +0x87\ngithub.com/grpc-ecosystem/go-grpc-middleware/recovery.UnaryServerInterceptor.func1.1(0x147f960, 0xc000596ea0, 0xc0004c61e0, 0xc0004aebc8)\n\t/home/jfd/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/recovery/interceptors.go:25 +0x92\npanic(0x10a3e40, 0x1c07850)\n\t/usr/local/go/src/runtime/panic.go:522 +0x1e0\ngithub.com/cs3org/reva/cmd/revad/svcs/grpcsvcs/interceptors/auth.NewUnary.func1(0x147f960, 0xc000596fc0, 0x116d640, 0xc000596d80, 0xc000426c80, 0xc0004cdd60, 0x0, 0x0, 0x0, 0x0)\n\t/home/jfd/Repositories/nexus/build/reva/src/cmd/revad/svcs/grpcsvcs/interceptors/auth/auth.go:143 +0xa55\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1(0x147f960, 0xc000596ea0, 0x116d640, 0xc000596d80, 0x0, 0x0, 0x0, 0x0)\n\t/home/jfd/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:34 +0x248\ngithub.com/grpc-ecosystem/go-grpc-middleware/recovery.UnaryServerInterceptor.func1(0x147f960, 0xc000596ea0, 0x116d640, 0xc000596d80, 0xc000426c80, 0xc0004cdd60, 0x0, 0x0, 0x0, 0x0)\n\t/home/jfd/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/recovery/interceptors.go:29 +0xf1\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1(0x147f960, 0xc000596ea0, 0x116d640, 0xc000596d80, 0x0, 0x0, 0x0, 0x0)\n\t/home/jfd/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:34 +0x248\ngithub.com/cs3org/reva/cmd/revad/svcs/grpcsvcs/interceptors/log.NewUnary.func1(0x147f960, 0xc000596ea0, 0x116d640, 0xc000596d80, 0xc000426c80, 0xc0004cdd60, 0x0, 0x0, 0x0, 0x0)\n\t/home/jfd/Repositories/nexus/build/reva/src/cmd/revad/svcs/grpcsvcs/interceptors/log/log.go:39 +0xc9\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1(0x147f960, 0xc000596ea0, 0x116d640, 0xc000596d80, 0x0, 0x0, 0x0, 0x0)\n\t/home/jfd/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:34 +0x248\ngithub.com/cs3org/reva/cmd/revad/svcs/grpcsvcs/interceptors/token.NewUnary.func1(0x147f960, 0xc000596ea0, 0x116d640, 0xc000596d80, 0xc000426c80, 0xc0004cdd60, 0x0, 0x0, 0x0, 0x0)\n\t/home/jfd/Repositories/nexus/build/reva/src/cmd/revad/svcs/grpcsvcs/interceptors/token/token.go:47 +0x36f\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1(0x147f960, 0xc000596e40, 0x116d640, 0xc000596d80, 0x0, 0x0, 0x0, 0x0)\n\t/home/jfd/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:34 +0x248\ngithub.com/cs3org/reva/cmd/revad/svcs/grpcsvcs/interceptors/appctx.NewUnary.func1(0x147f960, 0xc000596e40, 0x116d640, 0xc000596d80, 0xc000426c80, 0xc0004cdd60, 0x0, 0x0, 0x0, 0x0)\n\t/home/jfd/Repositories/nexus/build/reva/src/cmd/revad/svcs/grpcsvcs/interceptors/appctx/appctx.go:36 +0x32c\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1(0x147f960, 0xc000596d50, 0x116d640, 0xc000596d80, 0xc000426c80, 0xc000426ca0, 0x0, 0x0, 0x0, 0x0)\n\t/home/jfd/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:39 +0x24b\ngithub.com/cs3org/go-cs3apis/cs3/storageregistry/v0alpha._StorageRegistryService_GetStorageProvider_Handler(0x10c30c0, 0xc000252180, 0x147f960, 0xc000596d50, 0xc0004cdb30, 0xc0000a3a70, 0x0, 0x0, 0x0, 0x0)\n\t/home/jfd/go/pkg/mod/github.com/cs3org/go-cs3apis@v0.0.0-20190808160050-ce36ee374974/cs3/storageregistry/v0alpha/storageregistry.pb.go:374 +0x24d\ngoogle.golang.org/grpc.(*Server).processUnaryRPC(0xc00052a300, 0x14877e0, 0xc00052ad80, 0xc0003a3700, 0xc0002848d0, 0x1c0ae60, 0x0, 0x0, 0x0)\n\t/home/jfd/go/pkg/mod/google.golang.org/grpc@v1.22.0/server.go:998 +0x1038\ngoogle.golang.org/grpc.(*Server).handleStream(0xc00052a300, 0x14877e0, 0xc00052ad80, 0xc0003a3700, 0x0)\n\t/home/jfd/go/pkg/mod/google.golang.org/grpc@v1.22.0/server.go:1278 +0x976\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000036d50, 0xc00052a300, 0x14877e0, 0xc00052ad80, 0xc0003a3700)\n\t/home/jfd/go/pkg/mod/google.golang.org/grpc@v1.22.0/server.go:717 +0xb5\ncreated by google.golang.org/grpc.(*Server).serveStreams.func1\n\t/home/jfd/go/pkg/mod/google.golang.org/grpc@v1.22.0/server.go:715 +0xb0\n" traceid=c8202c5e80d8837f5f9ec033bb041cc1
4:24PM ERR svcs/grpcsvcs/interceptors/log/log.go:69 > unary code=Internal end="09/Aug/2019:16:24:50 +0200" from=tcp://[::1]:55276 pid=3896864 pkg=grpcserver start="09/Aug/2019:16:24:48 +0200" time_ns=2057434397 traceid=c8202c5e80d8837f5f9ec033bb041cc1 uri=/cs3.storageregistryv0alpha.StorageRegistryService/GetStorageProvider user-agent=grpc-go/1.22.0
4:25PM ERR svcs/httpsvcs/handlers/log/log.go:114 > http end="09/Aug/2019:16:25:20 +0200" host=::1 method=PROPFIND pid=3896864 pkg=httpserver proto=HTTP/1.1 size=0 start="09/Aug/2019:16:23:43 +0200" status=500 time_ns=96688241041 traceid=c8202c5e80d8837f5f9ec033bb041cc1 uri=/remote.php/dav/files/aaliyah_abernathy/ url=/remote.php/dav/files/aaliyah_abernathy/
```

the gateway logs this:
```
3:54PM INF cmd/revad/svcs/grpcsvcs/interceptors/log/log.go:69 > unary code=OK end="09/Aug/2019:15:54:51 +0200" from=tcp://[::1]:49178 pid=3893518 pkg=grpcserver start="09/Aug/2019:15:54:51 +0200" time_ns=6104393 traceid=90bd9b866ab7faa112390f034ca22a64 uri=/cs3.authv0alpha.AuthService/GenerateAccessToken user-agent=grpc-go/1.22.0
3:54PM ERR cmd/revad/svcs/grpcsvcs/gatewaysvc/storageprovidersvc.go:232 > gatewaysvc: error finding storage provider error="gatewaysvc: error calling GetStorageProvider: rpc error: code = Internal desc = runtime error: invalid memory address or nil pointer dereference" pid=3893518 pkg=grpcserver traceid=90bd9b866ab7faa112390f034ca22a64
```

I configured all http services to use the gateway, which is a separate process running on `localhost:1000`. Also can't get it to work by leaving out the gateway and setting:
```
[http.services.ocdavsvc]
gatewaysvc = "localhost:9999"
...
```

hm

Reply:
On problem for sure is that the ocdav (and ocs) service does nod use the storage registry to look up the storage provider. however, by configuring the storageprovider directly I should have been able to get a working connection ...